utils/githooks/pre-commit

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# A git hook to validate  and correct the copyright date in source files.
+
+regex='(^\s*\*.*)((Copyright\s*)([0-9]{4})(-([0-9]{4}))?)(\s*(Intel.*$))'
+year=$(date +%Y)
+errors=0
+
+files=$(git diff --diff-filter=AM --name-only --cached -- '*.c' '*.h' 'Makefile')
+for file in $files; do
+	read -r y1 y2 <<< "$(sed -nre "s/^.*$regex.*$/\4 \6/p" "$file")"
+	if [[ -z $y1 ]] ; then
+		echo "Copyright Information not found in: $file"
```

Comment:
Should we add a copyright here?

Reply:
I'll leave that to future improvements.   It's tricky because it needs to know what type of file it's dealing with.